### ISSUE TITLE
Emails, Import, Quick Search for Assigned To and Related to not working

### ISSUE THREAD

**Issue by DanVancPdx** [salesagility/SuiteCRM#6321]
#### Issue
Open Email and try to import it to an Account or Contact not working.  More specifically - the quick search to lookup an Account or Contact not working.

In the same form, the Assigned To quick search does work (it brings up a popup with available users to assign to), but, when a user is selected in the popup, it is not passed back to the form.

I am running a new Install 7.10.8.  (and this functionality did work in version 7.10.7)

#### Expected Behavior
I should be able to assign an email to a contact or an account using the IMPORT functionality in EMAIL.

#### Actual Behavior
The POPUP is not working.  There are two faults.  

1) Quick Search is not working for the account, case, contact fields
2) Quick Seach is working for looking up a "Assigned User", but when a user is selected in the popup, it is not passed back to the form.

#### Possible Fix
Not a fix, but here is some information that may help.

In Chrome, when I click on the "arrow" button to request the lookup, the Chrome console throws the following error:

```
Uncaught TypeError: Cannot read property 'parent_name' of undefined
    at HTMLSelectElement.onchange (index.php?module=Emails&action=index&parentTab=All:1)
```

#### Steps to Reproduce

1. You will need email SMTP incoming and outgoing setup, so that you can use the EMAIL module.
2. Click on Email so you can see your emails.  Click View Emails if necessary.
3. Using the suite select boxes on the left of your emails, select an email
4. Click "Bulk Action" and select "Import"
5. First Error: Click the "lookup arrow" to the right of "Assigned To".  Select any user in the POPUP.  When you select the user, it should be returned to the form.
6. Second Error: After the "Related to" label, click any module in the dropdown and click the "lookup arrow" to the right of "Related To".  You should get a POPUP window (but I dont).

#### Context
I'm just trying to attach email to customers and contacts using the IMPORT function of the EMAIL module.

#### Your Environment

* SuiteCRM Version used: 7.10.8
* Browser name and version up to date version of Chrome: Version 69.0.3497.92 (Official Build) (64-bit)
* Environment name and version: Mysql, PHP 7.0
* Operating System and version: 4.4.126-0.ksplice-lts-generic GNU/Linux

**Comment by DanVancPdx** [comment]
In SuiteCRM forum, user 3boptic has made a video showing this error, and a work-around.

SuiteForum link: [](url)https://suitecrm.com/suitecrm/forum/feedback/20392-import-emails-not-working-on-v7-10-8
Video [](url)https://drive.google.com/file/d/1KpXtp2PMakLiFcM2p9q4PE6sAiJfwRgJ/view?usp=sharing

**Comment by gymad** [comment]
related to issue #6050

**Comment by samus-aran** [comment]
Hi @DanVancPdx, Are you able to test the proposed fix and confirm if it helps resolves your issue. That would be much appreciated!

Thanks!

**Comment by DanVancPdx** [comment]
Partial fix.  With this fix, the quick search works with both the Assigned To and the Related To.  

In other words, I can select a new account and assigned.

But the Ok button doesn't save the selected account/assigned-to.

In fact, when I click on "OK", the ok/cancel row disappears, and the changes do not get applied, and the screen "Importing Email" stays showing instead of disappearing.

In this image below, the quick search found a customer and replaced it on the screen.  

![aaaaa](https://user-images.githubusercontent.com/31054514/45700480-86855500-bb21-11e8-9a69-24e189d7fecd.jpg)

Yay!  But then I clicked "OK", I received this:  

![bbbbb](https://user-images.githubusercontent.com/31054514/45700541-acaaf500-bb21-11e8-8228-818225561e2b.jpg)

The "OK/cancel" disappear, but the IMPORTING email screen should of applied the change, and the popup screen itself disappear.

This may be a second problem issue.  The original issue was that the quick search wasn't working.  Now the form isn't updating.  I'm not sure if this should be a new issue or stay in this thread.  (I'm new to github and am grateful for assistance).

**Comment by labcerouno** [comment]
I'm having a different behavior. After applying this fix, quicksearch won't work yet, but selection user from pop-up window (arrow button) will work.

**Comment by Dillon-Brown** [comment]
Hi @labcerouno @DanVancPdx, I've pushed a new commit to #6323, would you be able to confirm if this resolves your issue? (Make sure to clear cache, repair and rebuild and Rebuild Minified JS Files + Rebuild JS Grouping Files).

**Comment by DanVancPdx** [comment]
See my post above starting with Partial Fix.  

The proposed fix does allow the popup form to do a quick search and populate the fields in the form, however, the ok/cancel buttons disappear.

See the post above for screenshots.

Thanks for all the help!!!!

**Comment by labcerouno** [comment]
Thank you @Dillon-Brown! I had performed Repair and Rebuild, not Rebuild Minified JS + Grouping Files. Anyway, after performing these 3 tasks, the results are the same for me: no quicksearch, but I can click the arrow button and select an user from the pop-up window.

**Comment by labcerouno** [comment]
Thank you @Dillon-Brown ! 

After applying this new fix and repairing and rebuilding JS files, I'm in a worse scenario: quicksearch won't work yet. Clicking in arrow button and selecting from pop-up will fill the user field with the one I chose, but when clicking OK it happens the same thing as described by @DanVancPdx: the OK button disappears and the modal won't close. Importing never happens.

**Comment by labcerouno** [comment]
While testing the issue with emails imported automatically (which cannot be edited or changed the user assigned to), I activated the option "Import automatically" for the inbound group account I use. After this, I'm having the same behavior I had after the first commit: it will import only when I choose the user from pop-up window. The OK button in the modal is working again. \o/

**Comment by Dillon-Brown** [comment]
@labcerouno @DanVancPdx So far i'm unable to replicate, have you tried doing a hard refresh of the browser cache? this resolves the missing OK button and the emails correctly import for me (Using the latest version of the fix).

**Comment by Dillon-Brown** [comment]
@labcerouno I think we should seperate some of these issues out. If we can confirm that the proposed fix works for relating and importing emails then I would suggest creating a new issue for automatic importation. I've also seen an issue where there is no notification when importing the same email twice, and another where the pagination doesn't take you to the last page.

I think we have a few more minor bugs to fix before we can get this one fully closed out.

**Comment by labcerouno** [comment]
On v.7.10.9 OK button is working. Quicksearch is not. I can choose an user to assign to from the pop-up window after clicking the arrow button.

**Comment by labcerouno** [comment]
Hi @pgorod ! How could I test that? 

When I'm on Edit page (after importing a message) the quicksearch works. It seemed to be an issue with the modal. Watch below:
![beaob0nrt1](https://user-images.githubusercontent.com/2539666/45900530-74701480-bdb6-11e8-9505-937ec575ebf8.gif)

**Comment by labcerouno** [comment]
Applying #6013 won't fix this buggy behavior.

**Comment by Abuelodelanada** [comment]
Hi @labcerouno @Dillon-Brown @pgorod @DanVancPdx @samus-aran !!

As far I could investigate, the commit that introduced the error is:

https://github.com/salesagility/SuiteCRM/pull/6076/commits/5a8d6a7a1080a0664ef204361965870da769191f

If you revert this commit, everything seems to work ok:
I tested in 7.10.9


![6321](https://user-images.githubusercontent.com/939888/46499426-71354980-c7f6-11e8-8a09-23208e24b744.gif)

**Comment by Dillon-Brown** [comment]
Hi @Abuelodelanada, this should be fixed in #6323. The issue as you can see was that the formName was set to "EditNonImported" in that single file but hadn't been updated everywhere else.

**Comment by labcerouno** [comment]
According to what @Dillon-Brown suggested, I created a new issue for something discussed here: https://github.com/salesagility/SuiteCRM/issues/6440

**Comment by nettags** [comment]
Found a solution to get QuickSearch working again in 7.10.11:

Change form ID from "EditView" to "EditNonImported" in /modules/Emails/include/ImportView/ImportView.tpl:

From: <form class="email-import-view" id="EditView" name="EditNonImported" 
To: <form class="email-import-view" id="EditNonImported" name="EditNonImported"

**Comment by pgorod** [comment]
@nettags this issue is already closed and marked as fixed.

Exactly which problem (steps to reproduce) are you using that workaround for?

does it have anything to do with the related issue #6440?

Thanks!

**Comment by nettags** [comment]
Oh yes, you are right, it is fixed with commit 53774af by gymad - sorry for the confusion.
